server/src-lib/Hasura/RQL/DDL/Schema.hs

```diff
@@ -55,25 +58,47 @@ import           Hasura.Server.Utils            (matchRegex)
 
 data RunSQL
   = RunSQL
-  { rSql                      :: Text
-  , rCascade                  :: !(Maybe Bool)
-  , rCheckMetadataConsistency :: !(Maybe Bool)
+  { rSql                          :: Text
+  , rCascade                      :: !Bool
+  , rCheckMetadataConsistency     :: !(Maybe Bool)
+  , rTxAccessMode                 :: !Q.TxAccess
   } deriving (Show, Eq, Lift)
-$(deriveJSON (aesonDrop 1 snakeCase){omitNothingFields=True} ''RunSQL)
+
+instance FromJSON RunSQL where
+  parseJSON (Object o) = do
+    rSql <- o .: "sql"
+    rCascade <- o .:? "cascade" .!= False
+    rCheckMetadataConsistency <- o .:? "check_metadata_consistency"
+    isReadOnly <- o .:? "read_only" .!= False
+    let rTxAccessMode = if isReadOnly then Q.ReadOnly else Q.ReadWrite
+    pure RunSQL{..}
+  parseJSON _ = fail "expecting an object for args"
```

Comment:
This is what [`Data.Aeson.withObject`](https://hackage.haskell.org/package/aeson-1.4.6.0/docs/Data-Aeson.html#v:withObject) is for. :)